e2e/codecept.presettings.js

```diff
@@ -2,75 +2,70 @@ const waitOn = require('wait-on');
 const path = require('path');
 const execa = require('execa').command;
 
-const apps = {
-    ilc: {
-        port: 8233
-    },
-    registry: {
-        port: 4001
-    },
-    navigation: {
-        port: 8235
-    },
-    people: {
-        port: 8236
-    },
-    planets: {
-        port: 8237
-    },
-    fetchWithCache: {
-        port: 8238
-    },
-    news: {
-        port: 8239
-    },
-    system: {
-        port: 8240
-    }
+const appPorts = {
+    ilc: 8233,
+    registry: 4001,
+    navigation: 8235,
+    people: 8236,
+    planets: 8237,
+    fetchWithCache: 8238,
+    news: 8239,
+    system: 8240,
 };
 
-const ports = [
-    apps.navigation.port,
-    apps.people.port,
-    apps.planets.port,
-    apps.news.port,
-    apps.system.port,
-    apps.fetchWithCache.port,
-    apps.ilc.port,
-    apps.registry.port,
+const resources = [
+    `http-get://127.0.0.1:${appPorts.navigation}`,
+    `http-get://127.0.0.1:${appPorts.people}`,
+    `http-get://127.0.0.1:${appPorts.planets}`,
+    `http-get://127.0.0.1:${appPorts.news}/_spa/dev/assets-discovery`,
+    `http-get://127.0.0.1:${appPorts.system}/index.js`,
+    `http-get://127.0.0.1:${appPorts.fetchWithCache}`,
+    `http-get://127.0.0.1:${appPorts.ilc}/ping`,
+    `http-get://127.0.0.1:${appPorts.registry}/ping`,
 ];
 
 let childProcess;
 
 const shutDown = (done) => {
+    if (!childProcess) {
+        return done();
+    }
+
     console.log(`Killing child process...`);
-    childProcess.on('exit', () => {
+    childProcess.once('exit', () => {
         console.log(`Child process was successfully killed. Shutting down...`);
         done();
     });
-    childProcess.kill('SIGINT', { forceKillAfterTimeout: 3000 });
+    childProcess.kill('SIGTERM', { forceKillAfterTimeout: 3000 });
 };
 
 const bootstrap = async (done) => {
     try {
         await new Promise((resolve, reject) => {
-            childProcess = execa(`npm run start`, { cwd: path.join(__dirname, '..') });
-            childProcess.on('error', (error) => {
+            const verbose = process.env.TEST_ENV === 'verbose';
+            const stdio = verbose ? 'inherit' : 'ignore';
+
+            console.log('Launching ILC with demo apps for E2E tests...');
+            childProcess = execa(`npm run start -- --no-watch`, { cwd: path.join(__dirname, '..'), stdio });
+            childProcess.once('error', (error) => {
                 reject(error);
             });
+            childProcess.once('exit', () => !childProcess.killed && done(new Error('Child process exited...')));
 
             waitOn({
-                resources: ports.map((port) => `tcp:localhost:${port}`),
-                timeout: 60 * 1000,
+                resources,
+                timeout: 10 * 60 * 1000,
```

Comment:
Are you sure that we need to have such a timeout?

Reply:
Yeah, at CI we're pulling Docker image with demo apps so it may take a couple of minutes...